src/Goetas/Twital/EventSubscriber/IDNodeSubscriber.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace Goetas\Twital\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Goetas\Twital\EventDispatcher\TemplateEvent;
+use Goetas\Twital\Twital;
+
+/**
+ *
+ * @author Asmir Mustafic <goetas@gmail.com>
+ *
+ */
+class IDNodeSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return array(
+            'compiler.post_load' => array(
+                array(
+                    'addAttribute'
+                )
+            ),
+            'compiler.pre_dump' => array(
+                array(
+                    'removeAttribute'
+                )
+            )
+        );
+    }
+
+    public function addAttribute(TemplateEvent $event)
+    {
+        $doc = $event->getTemplate()->getDocument();
+        $xp = new \DOMXPath($doc);
+        $nodes = $xp->query("//*[@*[namespace-uri()='".Twital::NS."']]");
+        $cnt = 0;
+        foreach ($nodes as $node) {
+            $node->setAttributeNS(Twital::NS, '__internal-id__', ++$cnt);
```

Comment:
> @soukiii Oh, I get it now. First part of hash is unique for current document and second part ensures that it won't colide with normal variables.

My first idea was to assign a random string instead of `++$cnt`. Doing this I will be more safe with multiple documents... What is your opinion about this?